src/main/java/com/iota/iri/service/milestone/impl/LatestSolidMilestoneTrackerImpl.java

```diff
@@ -94,6 +96,17 @@
      */
     private TransactionRequester transactionRequester;
 
+    /**
+     * The actual start milestone index from which the node started from
+     */
+    private int milestoneStartIndex;
```

Comment:
Can you maybe remove this field due to two reasons:
1. Once it is here we get another object that is a source of `milestoneStartIndex`. Sometimes later another dev will come along and add a getter for this. Eventually the design will become more spaghetti. The way we try to prevent spaghetti design is by having all leaving "service" class stateless by depend on other stateless "service" classes, and having data come from data providers so that we have one source of truth.

2. IIUC, you only initialize this field once when the tracker service first runs. So if the node comes out of sync again but doesn't get shut down (e.g. internet connection was lost), then the next time the starting point will be wrong... so I think this is a bug

Reply:
Variables are now in an inner class.